src/protocol/cEMI/CEMIMessage.ts

```diff
@@ -0,0 +1,41 @@
+import KNXAddress from "../KNXAddress";
+import KNXDataBuffer from "../KNXDataBuffer";
+import ControlField from "./ControlField";
+import NPDU from "./NPDU";
+
+export default class CEMIMessage {
+  msgCode: number;
+  length: number;
+  additionalInfo: KNXDataBuffer | null;
+  control: ControlField;
+  srcAddress: KNXAddress;
+  dstAddress: KNXAddress;
+  npdu: NPDU;
+
+  constructor(msgCode: number, length: number, additionalInfo?: KNXDataBuffer, control?: ControlField, srcAddress?: KNXAddress, dstAddress?: KNXAddress, npdu?: NPDU) {
+    this.msgCode = msgCode;
+    this.length = length;
+    this.additionalInfo = additionalInfo;
+    this.control = control;
+    this.srcAddress = srcAddress;
+    this.dstAddress = dstAddress;
+    this.npdu = npdu;
+  }
+
+  toBuffer(): Buffer {
+    const buffer = Buffer.alloc(2);
+    buffer.writeUInt8(this.msgCode, 0);
+    const len = this.additionalInfo == null ? 0 : this.additionalInfo.length;
+    buffer.writeUInt8(len, 1);
+    if (this.additionalInfo) {
+      return Buffer.concat([buffer, this.additionalInfo.value]);
```

Comment:
@Supergiovane qui era `this.additionalInfo` ed ho cambiato mettendo `this.additionalInfo.value`, perchè mi aspetto che `this.additionalInfo` sia di tipo `KNXDataBuffer`, corretto?